.pre-commit-config.yaml

```diff
@@ -0,0 +1,19 @@
+# Minimum version of pre-commit
+minimum_pre_commit_version: "2.9.0"
+
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.6.0
+    hooks:
+      # Prevent committing directly to the main branch
+      - id: no-commit-to-branch
+        args: [--branch, main]
+        stages: [commit]
+    
+  - repo: local
+    hooks:
+      - id: clang-format
+        name: Run git-clang-format
+        entry: git clang-format --style=file --staged
```

Comment:
As written, this will format all lines of code in any staged file. We should only format the lines which were changed by this commit. Add `--diff $(git merge-base HEAD main)` to the command (taken from `.github/workflows/pr-format-check.yml`)